### PR TITLE
fix(api) Use the web service or initialize it

### PR DESCRIPTION
**Pull request by loiclau**
<h1> Pull Request Template </h1>

<h2> Description </h2>

Use the existing web service or initialize it.
Need to use web service with cron script.

<h2> Type of change </h2>

- [x] New functionality (non-breaking change)

<h2> Target serie </h2>

- [x] 19.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

<h2> Checklist </h2>

<h5> Community contributors & Centreon team </h5>

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/api/class/webService.class.php [hunk 1]
```diff
@@ -305,8 +305,13 @@ public static function router(\Pimple\Container $dependencyInjector, $user, $isI
             // Initialize the language translator
             $dependencyInjector['translator'];
 
-            $wsObj = new $webService['class'];
-            $wsObj->setDi($dependencyInjector);
+            // Use the web service if it is initialize or initialize it
```
Comment (sc979) [inline www/api/class/webService.class.php]:
```suggestion
            // Use the web service if has been initialized or initialize it
```

**Comment by vhr** [comment]
@loiclau can you explain the idea behind the changes?

**Comment by loiclau** [comment]
It's because the webservice need a repository to work. 
It's add and initialized on the service provider.
Without this change we can not create objects with argument.

**Comment by vhr** [comment]
You can register your webservice here

https://github.com/centreon/centreon/blob/master/src/Centreon/ServiceProvider.php#L37

example:
`$pimple['centreon.webservice']->add(Webservice\MyWebservice::class);`

about calling of repositories

```
class MyWebservice extends CentreonWebServiceAbstract
{
    public function getSomeMethod() {
		$entities = $this->getDi()['centreon.db-manager']
			->getRepository(Repository\MyRepository::class)
			->someMethodFromRepository();

		...

		return [];
	}

	...
}
```

**Pull request merged**